pandaharvester/harvesterpreparator/analysis_aux_preparator.py

```diff
@@ -139,7 +141,9 @@ def trigger_preparation(self, jobspec):
                 executionID = None
                 if return_code == 0 and 'check' in bulkExtCommand[protocol]['command']:
                     executionID = [s for s in stdout.split('\n') if s][-1]
-                    executionID = '{0}:{1}'.format(protocol, executionID)
+                    dst = bulkExtCommand[protocol]['dst'][0]
```

Comment:
If you want to check destinations of all files shouldn't it be

`
','.join(bulkExtCommand[protocol]['dst'])
`

Reply:
That step is done in line 128. At this point I am just getting execution id which includes
the path.

Reply:
The string in L128 is given to the command to trigger preparation, while the string in L142 is given to the command to check preparation via executionID at L193. If the job submits a batch job for two files, why is it enough to check only one destination?

Reply:
The checking code only processes one file at a time. 
Here is the routine - It assumes that only one container is check at a time
------------
#!/bin/sh

# This shell script takes as inputs 

HARVESTER_DIR=/gpfs/alpine/hep113/world-shared/harvester
CONTAINER_DIR=${HARVESTER_DIR}/var/lib/containers


function err() {
  dt=$(date --utc +"%Y-%m-%d %H:%M:%S,%3N [wrapper]")
  echo $dt $@ >&2
}

function log() {
  dt=$(date --utc +"%Y-%m-%d %H:%M:%S,%3N [wrapper]")
  echo $dt $@
}

function usage () {
  echo "Usage: $0 -i <destination path>"
  echo
  echo "  -i,   detination path" 
  echo
  exit 1
}

function main() {

  # full path to container
  CONTAINER_DIR=$iarg
  # test if container image exists
  if [ "$(ls -A $CONTAINER_DIR)" ]; then
      echo "Skip creating container $CONTAINER_DIR is not Empty"
      exit 0
  else
      echo "$CONTAINER_DIR is Empty"
      exit 1
  fi
}

starttime=$(date +%s)


iarg=''

POSITIONAL=()
while [[ $# -gt 0 ]]
do
key="$1"
case $key in
    -h|--help)
    usage
    shift
    shift
    ;;
    -i)
    iarg="$2"
    shift
    shift
    ;;
esac
done

if [ -z "${iarg}" ]; then usage; exit 1; fi


main

Reply:
The code is not only for containers. It is quite generic so that it can be configured to execute any command for any protocol for one or more files. The point is that the code should not assume only your script and/or one particular usecase.

In the first place, if len(bulkExtCommand[protocol]['dst']) == 1, 
','.join(bulkExtCommand[protocol]['dst']) == bulkExtCommand[protocol]['dst'][0]
so it will work with your script as well.